### ISSUE TITLE
PCTCompile - Delete warning file if empty

### ISSUE THREAD

**Issue by clement-brodu** [Riverside-Software/pct#354]
### Subject of the issue
I use the -swl option to hide the warning 5410 (-D increment).

I have not the warning in the `.warnings` file (good) but the `.warnings` is not deleted.
Is there a reason to keep an empty warnings file ?

Thank you, 

### Your environment
* Operating Windows 10
* OpenEdge 11.7.4
* Ant 1.10.5
* PCT 211

### Actual behaviour
PCT keep an empty warnings file

### Expected behaviour
If empty, the warning file should be remove.

### Steps to reproduce
In PCTCompile, use the -swl option to ignore a warning.
```
<Option name="-swl" value="swl.txt" />
```

For exemple, put a RETURN somewhere in a middle of a function, method... :
> WARNING: Executable statement at line <line-num> of file <file-name> will not be reached. (15090)

**Comment by gquerret** [comment]
I agree that the `.warnings` file should be removed when there are no more warnings in a given file. But I don't see the link between this bug and the -swl option to ignore message 5410. Is this message reported in the .warnings file ?

**Comment by clement-brodu** [comment]
Without the -swl option, the probleme can't happen becauses all warning are logged.

But with the -swl, COMPILER:WARNING is true, so the IF condition is True, so the file is created.
Then, `(LOOKUP(STRING(COMPILER:GET-NUMBER(i)), SESSION:SUPPRESS-WARNINGS-LIST) EQ 0)` avoid to log the warning, but the file is still there.

```abl
IF COMPILER:WARNING OR lWarnings THEN DO:
      OUTPUT STREAM sWarnings TO VALUE(warningsFile).
      DO i = 1 TO COMPILER:NUM-MESSAGES:
        IF bAbove101 THEN DO:
          /* Pointless message coming from strict mode compiler */
          IF COMPILER:GET-NUMBER(i) EQ 2411 THEN NEXT.
          /* Messages 2363, 3619 and 3623 are in fact warnings (from -checkdbe switch) */
          IF (COMPILER:GET-MESSAGE-TYPE(i) EQ 2) OR (COMPILER:GET-NUMBER(i) EQ 2363) OR (COMPILER:GET-NUMBER(i) EQ 3619) OR (COMPILER:GET-NUMBER(i) EQ 3623) THEN DO:
            IF (LOOKUP(STRING(COMPILER:GET-NUMBER(i)), SESSION:SUPPRESS-WARNINGS-LIST) EQ 0) THEN DO:
              PUT STREAM sWarnings UNFORMATTED SUBSTITUTE("[&1] [&3] &2", COMPILER:GET-ROW(i), REPLACE(COMPILER:GET-MESSAGE(i), '~n', ' '), COMPILER:GET-FILE-NAME(i)) SKIP.
            END.
          END.
        END.
      END.
      FOR EACH ttWarnings:
        PUT STREAM sWarnings UNFORMATTED SUBSTITUTE("[&1] [&3] &2", ttWarnings.rowNum, ttWarnings.msg, ttWarnings.fileName) SKIP.
      END.
      OUTPUT STREAM sWarnings CLOSE.
    END.
```

**Comment by gquerret** [comment]
I'll delete the file after this block if no warning has been written.
In the mean time, you can raise the initial value of -D, that's almost always beneficial for compilation.

**Comment by clement-brodu** [comment]
I'll try that, thanks for the advice.

**Comment by gquerret** [comment]
Fixed as part of #406
